frontend/packages/dev-console/src/components/topology/podring-utils.ts

```diff
@@ -0,0 +1,58 @@
+import * as _ from 'lodash';
+import { TransformTopologyData } from './topology-utils';
+
+export default class TransformPodRingData {
+  private podsData = {
+    pods: {},
+  };
+
+  private topologyDataTransform;
+
+  constructor(public resources) {
+    this.topologyDataTransform = new TransformTopologyData(resources);
+  }
+
+  public getPodsData() {
+    return this.podsData;
+  }
+
+  public transformDataBy = (targetDeployment = 'deployments'): TransformPodRingData => {
+    if (!this.topologyDataTransform.deploymentKindMap[targetDeployment]) {
+      throw new Error(`Invalid target deployment resource: (${targetDeployment})`);
+    }
+    if (_.isEmpty(this.resources[targetDeployment].data)) {
+      return this;
+    }
+
+    const targetDeploymentsKind = this.topologyDataTransform.deploymentKindMap[targetDeployment]
+      .dcKind;
+    const resourceData = this.resources[targetDeployment].data;
+
+    _.forEach(resourceData, (d) => {
+      d.kind = targetDeploymentsKind;
+      const dUID = _.get(d, 'metadata.uid');
+      const deploymentsLabels = _.get(d, 'metadata.labels') || {};
+      const replicationControllers = this.topologyDataTransform.getReplicationControllers(
+        d,
+        targetDeployment,
+      );
+      const current = _.head(replicationControllers);
+      const previous = _.nth(replicationControllers, 1);
+      const currentPods = current
+        ? this.topologyDataTransform.transformPods(this.topologyDataTransform.getPods(current, d))
+        : [];
+      const previousPods = previous
+        ? this.topologyDataTransform.transformPods(this.topologyDataTransform.getPods(previous, d))
+        : [];
+
+      this.podsData.pods[dUID] = {
+        id: dUID,
+        name: deploymentsLabels['app.kubernetes.io/instance'] || _.get(d, 'metadata.name'),
+
+        type: 'workload',
```

Comment:
Doesn't look like `id`, `name` or `type` are used anywhere.
Seems the data structure was just copied from topology instead of tailoring a new data structure strictly for the pod visualization.